### ISSUE TITLE
Modify our catch main to allow input files for tests

### ISSUE THREAD

**Issue by bartgol** [E3SM-Project/scream#140]
Currently, catch2 does not let you pass arguments to the executable. This is because any 'word' following the exec name is interpreted as the TEST_CASE to be run.

[This page](https://github.com/catchorg/Catch2/issues/793) shows how one can add this capability by modifying the command line interface of the catch2 Session.

A good solution is probably to add the flag `--scream-input-file`, which is basically guaranteed not to clash with future catch2 flags.

**Comment by ambrad** [comment]
My approach was probably hacky, but the tridiag solver testing exe serves as both a catch2 exe and a perf tester with cmd-line inputs.

**Comment by bartgol** [comment]
We could do as you did, namely read the argc/argv first, and *then* call the catch2 session, but we would not be able (I think) to then forward whatever the cmd line arg was to the actual test. Your hack works cause the perf test runs _outside_ of catch2.

**Comment by bartgol** [comment]
TBC, ultimately, I want to be able (during development, at least) to run the catch test with a few particular configurations, without having to change the source and recompile (and I don't want to loop over all configurations, or doing N runs with random inputs, hoping to get the config I want to test).

**Comment by bartgol** [comment]
Turns out both approaches are the same, cause there seem to be no way to access the catch Session from within a test (odd: Session makes sure there's only one instance of the class, and yet provides no singleton implementation, in the sense of a static way to access that single instance).

So at the end of the day, to fwd params to the tests, I needed to create a `TestSession` singleton class, storing whatever args we want to fwd to the test. I still prefer to do it through catch CLI though, since it is easy to do, and catch will include your option (along with any help string you provide) at the end of the output of `test_name -h`.

I will put up a quick PR soon.
